### PR TITLE
Ruff code formatting fixes

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
This PR updates the version of python used to run the [ruff](https://github.com/astral-sh/ruff) code formatting check (since python 3.8 is nearing end of life), and updates the code for compliance with ruff's linting specifications.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -53,5 +53,5 @@ ignore = [
     "SIM117", # flake8-simplify - some of merged with statements are not looking great with black, reanble after drop python 3.9
 ]
 
-target-version = "py38"
+target-version = "py312"
```
Comment (jni) [inline pyproject.toml:56]: I think this should be the minimum supported version, which right now is py310?

Comment (GenevieveBuckley) [inline pyproject.toml:56]:
```suggestion
target-version = "py39"
```

Comment (GenevieveBuckley) [reply pyproject.toml]: Ok, will change. Minimum python version will be python 3.9 after dropping python 3.8

**Comment by GenevieveBuckley** [comment]
We should be able to merge this after the CI fixes in https://github.com/napari/napari-animation/pull/222 are merged

**Comment by GenevieveBuckley** [comment]
Closed, in favour of https://github.com/napari/napari-animation/pull/224
This PR has more ruff updates than needed, since we decided to target ruff to python 3.9 instead of python 3.12

**Pull request closed without merging**